kickstarter/android-oss
# PAY-1895: Present PaymentSheet on user settings

# 📲 What

Present PaymentSheet on user settings

# 🤔 Why

Some background context on why the change is needed.

# 🛠 How
- In this task we only present the paymentSheet after callint `createsetupIntent` mutation, and handle the sucsess/error states for that network call

# 👀 See
-- Sucess present the paymentSheet

https://user-images.githubusercontent.com/4083656/189988694-26087437-a941-4b37-a61e-83549415eed4.mp4


-- Error present the error snackbar with a generic message

https://user-images.githubusercontent.com/4083656/189988811-a9a63972-acb0-4837-9e5f-3cb86af3f4e1.mp4

|  |  |

# 📋 QA
- Go to Settings -> account -> payment methods> hit add new card button. You should see the progress bar active and then the PaymentSheet presented, try pressing the "add new card button" several times
- Go to Settings -> account -> payment methods> turn on airplane mode. You should see the error snackbar as the network is not available.

# Story 📖

[NTV-1895](https://kickstarter.atlassian.net/browse/PAY-1895)
